### ISSUE TITLE
Suppress the folder path setup

### ISSUE THREAD

**Issue by nicucalcea** [ajrgodfrey/BrailleR#97]
Thank you for this excellent package.

I am a building a workflow to improve the accessibility of my plots, and I would like to use the `VI` function in `BrailleR` to automatically generate some boilerplate text that I can then edit to be more specific.

Since I plan on sharing this workflow with my colleagues and also put it up on GitHub, I was wondering if there is a way to suppress the path selection setup functions? Essentially, my package would only invoke the `VI` function on a plot, so it would be redundant to create a permanent folder, and I would like to avoid unnecessary steps for my users.

Let me know if this is possible, and if not, I'm happy to suggest a solution and/or submit a PR.

Thanks again!

**Comment by ajrgodfrey** [comment]
Perhaps you need to call the VI function without loading the package.

Graph = Data |> ggplot(...) + ...
BrailleR::VI(Graph)


I do need to caution use of the output for al text. The reason for creating a graph is what the blind user needs to know. It must be succinct and informative. My package is aimed at making sure a blind person can be more confident they are making a graph that their sighted audience will appreciate. That requires knowing what elements make up the graph in an objective sense. The subjective elements of a graph cannot be automated.

HTH,
Jonathan

**Comment by nicucalcea** [comment]
Thanks for the prompt reply, Jonathan!

I am already calling the function directly in my package. Not an expert on this, but I think that because the setup function [runs in `.onLoad`](https://github.com/ajrgodfrey/BrailleR/blob/21a4c072dbd167209da76c9e9fa2193e38e59e50/R/zzz.R#L20) rather than `.onAttach`, so it gets called every time any function from the package is invoked.

I understand your concern about the alt text. I am a journalist who works with data, so I completely agree about what the alt graph should be. In fact, my editors would not publish an alt text that is automated anyway.

My newsroom uses an R package that exports graphs in the right size and styling for our website. My plan is to add a step in the export that checks whether the plot has alt text declared in `labs(alt = "Alt text here")`, and if not, pop a warning in the console. I was thinking it would be good to have the result of `BrailleR::VI()` offered as a starting point. It would print something like this to the console:

```
✖ Did you forget to include alt text? You can do so in labs(alt = "This is some text describing the chart.").
→ Here's something to get you started: Output from BrailleR goes here
```

Hope that helps explain what I'm trying to accomplish. Obviously, this should be a low priority for you and not something worth changing if it interferes with how the package functions.

**Comment by HenrikBengtsson** [comment]
In `SetupBrailleR()` you ask the user for their permission to store files in `~/MyBrailleR/`. This is required because the CRAN Policies require the user's permission for storing files in user's home directory.  However, you can, _without_ permission, store files under:

```r
Folder = tools::R_user_dir("BrailleR", which = "config")
```

That way, you can avoid having the prompt in the first place.  That folder is permanent, and in the user's home folder.  On Linux, it defaults to `~/.config/R/BrailleR/`.

This is what the CRAN Policies says:

> For R version 4.0 or later (hence a version dependency is required or only conditional use is possible), packages may store user-specific data, configuration and cache files in their respective user directories obtained from `tools::R_user_dir()`, provided that by default sizes are kept as small as possible and the contents are actively managed (including removing outdated material).

Source:  https://cran.r-project.org/web/packages/policies.html

R 4.0.0 was release on 2020-04-28, so 3.5 years ago now.

**Comment by ajrgodfrey** [comment]
Useful reminder; thanks Henrik. Suspect it is a nice distraction from the main issue though.

I'm not going to meet the "the contents are actively managed" criterion.
I may also have to question total file size etc.


I think therefore that the opportunity to use the automatically  named folder might replace the temporary folder but I'd have to change a lot of functions to use it without breaking the rules on active maintenance. I had always hoped users would do things like make conversions of vignettes in HTML instead of pdf which could lead to the folder getting messy.

I do need users to get to the MyBrailleR folder which I guess could be resolved with a new MyBrailleR() function which just opens the folder no matter where it is.

This still makes a folder though, and the issue is whether we need to do so for use of BrailleR::VI() without library()

**Comment by hughjonesd** [comment]
If the MyBrailleR folder only needs to be created for certain functions, why not check for its existence only when those functions are called? That way you don't need to concern yourself with whether the package is attached or not. It seems like package attachment is an orthogonal issue.

**Comment by nicucalcea** [comment]
My concern is that people might find it confusing if they receive a prompt to set up a library they're not familiar with, but if it happens in the background, I think that would be okay.

**Comment by hughjonesd** [comment]
Right, but why's that better if you do it onLoad than on calling a function?
